public/app/features/alerting/unified/components/rule-editor/NotificationsStep.tsx

```diff
@@ -12,6 +15,10 @@ export const NotificationsStep = () => {
   const styles = useStyles2(getStyles);
   const theme = useTheme2();
 
+  const { watch } = useFormContext<RuleFormValues>();
+
+  const dataSourceName = watch('dataSourceName');
```

Comment:
With current changes , for grafana-managed alert rules you don't see the custom labels drop-down

<img width="1420" alt="Screenshot 2022-11-09 at 10 10 11" src="https://user-images.githubusercontent.com/33540275/200791310-b370f353-3db1-45af-a354-4567f94cca29.png">

this is because in case of grafana-managed alert rules this watch returns null, so, I think the correct dataSourceName should be:
`const dataSourceName = watch('dataSourceName') ?? GRAFANA_RULES_SOURCE_NAME`

Reply:
> With current changes , for grafana-managed alert rules you don't see the custom labels drop-down
> 
> <img alt="Screenshot 2022-11-09 at 10 10 11" width="1420" src="https://user-images.githubusercontent.com/33540275/200791310-b370f353-3db1-45af-a354-4567f94cca29.png">
> 
> this is because in case of grafana-managed alert rules this watch returns null, so, I think the correct dataSourceName should be: `const dataSourceName = watch('dataSourceName') ?? GRAFANA_RULES_SOURCE_NAME`

Great catch Sonia! 💯 You're right, for grafana managed alert rules we don't have the `dataSourceName` value. I agree it makes sense to set it ourselves if that's the case. Fix done 👍